### DISCUSSION TITLE
WindTerm 2.5.0 Prerelease 1

### DISCUSSION THREAD

**Discussion by kingToolbox** [kingToolbox/WindTerm#774] [General]
# Spotlight

### Auto completion suggests `commands`, `subcommands`, `command line arguments` and `command histories` that are currently available. The `command snippets` from the `QuickBar` are also available in the completion list. Auto completion greatly increases text input speed, because you don’t have to memorize a large number of commands and type every letter in a command.

**Auto completion supports:**
- **Linux Shell commands.**
- **MacOS Shell commands.**
- **Windows Cmd commands.**
- **PowerShell commands.**
- **Commands of any command line program, such as Git**
- **Command Snippets from the QuickBar**

# Usage:
### Windows

|   Version | Architecture |  Operating Systems |
| --------- | -------------- | --------------------- |
| WindTerm_2.5.0_Prerelease_1_Windows_Portable_x86_64 | x86_64 | Windows 7, Windows 8, Windows 10, Windows 11 |

#### Versions before Version 2.4.0

1. Start `WindTerm` and select the `profiles directory` and **quit**.
2. Replace the new `.wind/profiles` folder with the old `profiles` folder.
3. Restart `WindTerm` and all the sessions and configuration will be restored.

#### Versions after Version 2.4.0

1. Start `WindTerm` and select the `profiles directory` containing the old `.wind` folder.
2. It is recommended to choose the `Home Directory` to facilitate subsequent upgrades.

If the `.wind` folder needs to be moved to another system:
1. Quit `WindTerm`.
4. Replace the new `.wind` folder with the old `.wind` folder.
5. Restart `WindTerm` and all the sessions and configuration will be restored.

The **.wind** folder is cross-platform. 

# Known Issues:
- [ZMODEM] **Only support file names contains ASCII characters in Windows, otherwise the file name of the transfered file will be garbled after transmission, but it does not affect the correctness of the file content.** Linux and MacOS version does not have this problem.

# Changes in Version 2.5.0 Prerelease 1 (2022-5-18):

# Features:
- [SESSION] **Supports auto completion for `Linux Shell`, `MacOS Shell`, `PowerShell`, `Windows Cmd`, `Git` and so on.** [Intro video](https://kingtoolbox.github.io/2022/05/18/auto_completion/)
  - **22** Git Commands
  - **32** Windows Cmd Commands
  - **42** Linux / MacOS Shell Commands
  - **3695** PowerShell Commands of `Windows Server 2022` and `PowerShell v7.3`
- [SESSION] **Supports auto completion for the command snippets from the `QuickBar`.**
- [GUI] **The filtering algorithm of the filter boxes is modified to a `multiple pattern matching algorithm`**.
- [TAB] Restore the last modified tab name. #626 
- [FILER] Add menu items `Copy directory path` and `Copy directory path to terminal`.

# Improvements:
- [SCHEME] `Cmd` supports highlighting `Slash Options`.
- [SESSION] The default lines of scrollback history is limited to `999999`.

# Bugs:
- [SESSION] `Clink` fails to start when the application path contains spaces. #736 #746

# Intro Videos
- [Auto Completion](https://kingtoolbox.github.io/2022/05/18/auto_completion/)

<hr /><em>This discussion was created from the release <a href='https://github.com/kingToolbox/WindTerm/releases/tag/2.5-prerelease'>WindTerm 2.5.0 Prerelease 1</a>.</em>

**Comment by as4lover** [comment]
Hi,
Thank you for updating this program.
I would like to know how to disable "Auto Completion" option.

**Reply by kingToolbox** [reply]
Hello, please download [WindTerm_2.5.0_Prerelease_3](https://github.com/kingToolbox/WindTerm/releases/tag/2.5-prerelease) and open `Menubar- Preferences - Settings Dialog - Terminal - Auto Completion` to disable it.

**Comment by katlqa** [comment]
感谢！

**Comment by zyystudio** [comment]
2.x开始用这个神器，非常棒！就是一直想关掉2.5的autocomplete，今天prerelease3终于开心的继续用起来了。

**Reply by kingToolbox** [reply]
Sorry, I know that autocompletion not only fails to help, but also brings a lot of trouble to use. But can you describe in a little more detail what makes you uncomfortable when you use it? Such as the command you expect to enter, and the candidates provided by autocomplete at this time and so on.

I'd love to improve autocompletion further, after all I still think it's a useful feature.

**Reply by zyystudio** [reply]
Sure. For example, 
I use "ls -ltar" quite often. It sometimes automatically gives me "less". I type so quickly that I can not remember which character I am typing lets it think it might be "less".
I fully agree this feature is useful for new bees in one new environment. For me it might be helpful when dealing with some non linux environment like powershell command autocomplete. 
Keep evolving! Love you!

**Comment by katlqa** [comment]
whte



| |
夏天秋水
|
|
***@***.***
|




---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2022年07月13日 13:20 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.******@***.***> |
| 主题 | Re: [kingToolbox/WindTerm] WindTerm 2.5.0 Prerelease 1 (Discussion #774) |

2.x开始用这个神器，非常棒！就是一直想关掉2.5的autocomplete，今天prerelease3终于开心的继续用起来了。

**Comment by darkhandz** [comment]
Before Pre 3, I also want to find a way to disable the Auto Completion feature, but after I updated to Pre 3 and test for a while, I feel this feature is usable now.
